components/butterfly/src/server/expire.rs

```diff
@@ -41,6 +41,14 @@ fn run_loop(server: &Server, timing: &Timing) -> ! {
                   .start_hot_rumor(RumorKey::new(RumorType::Member, &id, ""));
         }
 
+        let now = Utc::now();
+        server.departure_store.purge_expired(now);
+        server.election_store.purge_expired(now);
+        server.update_store.purge_expired(now);
+        server.service_store.purge_expired(now);
+        server.service_config_store.purge_expired(now);
+        server.service_file_store.purge_expired(now);
+
```

Comment:
I wonder what kind of impact this kind of frequent operation on our shared data structures will have (if any).

If there's a significant impact, it may be worth reducing the frequency of these purges.

Reply:
Yeah, doing this every 500ms feels like overkill for sure, but this felt like the obvious place to put this code.  I suppose we could always spawn a separate thread, just for rumor expiration, with a much longer sleep time.

Reply:
I opted not to spawn a separate thread for this, but instead created a counter of sorts inside this thread, so that the purging only happens every 60 seconds, configurable via an environment variable.

Reply:
This definitely strikes me as the sort of thing we won't really know which way to go on without some field experience.

Right now, I think the efficiency should probably be ok since we're not ever shrinking any of the collections' allocations. However, it probably is the case that we should do that every so often, lest our memory utilization be dictated by the highest point of concurrently live rumors.